pkg/controller/scyllacluster/resource/resource_test.go

```diff
@@ -314,3 +321,590 @@ func TestMemberService(t *testing.T) {
+				return s
+			}(),
+			expectedError: nil,
+		},
+		{
+			name: "new StatefulSet with non-nil NodeAffinity",
+			rack: func() scyllav1.RackSpec {
+				r := basicRack
+				r.Placement = &scyllav1.PlacementSpec{
+					NodeAffinity: newNonNilNodeAffinity(),
+				}
+				return r
+			}(),
+			scyllaCluster:       newBasicScyllaCluster(),
+			existingStatefulSet: nil,
+			sidecarImage:        basicSidecarImage,
+			expectedStatefulSet: func() *appsv1.StatefulSet {
+				s := newBasicStatefulSet()
+				s.Spec.Template.Spec.Affinity.NodeAffinity = newNonNilNodeAffinity()
+				return s
+			}(),
+			expectedError: nil,
+		},
+		{
+			name: "new StatefulSet with non-nil PodAffinity",
+			rack: func() scyllav1.RackSpec {
+				r := basicRack
+				r.Placement = &scyllav1.PlacementSpec{
+					PodAffinity: newNonNilPodAffinity(),
+				}
+				return r
+			}(),
+			scyllaCluster:       newBasicScyllaCluster(),
+			existingStatefulSet: nil,
+			sidecarImage:        basicSidecarImage,
+			expectedStatefulSet: func() *appsv1.StatefulSet {
+				s := newBasicStatefulSet()
+				s.Spec.Template.Spec.Affinity.PodAffinity = newNonNilPodAffinity()
+				return s
+			}(),
+			expectedError: nil,
+		},
+		{
+			name: "new StatefulSet with non-nil PodAntiAffinity",
+			rack: func() scyllav1.RackSpec {
+				r := basicRack
+				r.Placement = &scyllav1.PlacementSpec{
+					PodAntiAffinity: newNonNilPodAntiAffinity(),
+				}
+				return r
+			}(),
+			scyllaCluster:       newBasicScyllaCluster(),
+			existingStatefulSet: nil,
+			sidecarImage:        basicSidecarImage,
+			expectedStatefulSet: func() *appsv1.StatefulSet {
+				s := newBasicStatefulSet()
+				s.Spec.Template.Spec.Affinity.PodAntiAffinity = newNonNilPodAntiAffinity()
+				return s
+			}(),
+			expectedError: nil,
+		},
+		{
+			name: "new StatefulSet with non-nil ImagePullSecrets",
+			rack: basicRack,
+			scyllaCluster: func() *scyllav1.ScyllaCluster {
+				sc := newBasicScyllaCluster()
+				sc.Spec.ImagePullSecrets = newNonNilImagePullSecrets()
+				return sc
+			}(),
+			existingStatefulSet: &appsv1.StatefulSet{},
+			sidecarImage:        basicSidecarImage,
+			expectedStatefulSet: func() *appsv1.StatefulSet {
+				sts := newBasicStatefulSet()
+				sts.Spec.Template.Spec.ImagePullSecrets = newNonNilImagePullSecrets()
+				return sts
+			}(),
+			expectedError: nil,
+		},
+		{
+			name: "new StatefulSet with non-nil ForceRedeploymentReason",
+			rack: basicRack,
+			scyllaCluster: func() *scyllav1.ScyllaCluster {
+				sc := newBasicScyllaCluster()
+				sc.Spec.ForceRedeploymentReason = "reason"
+				return sc
+			}(),
+			existingStatefulSet: &appsv1.StatefulSet{},
+			sidecarImage:        basicSidecarImage,
+			expectedStatefulSet: func() *appsv1.StatefulSet {
+				sts := newBasicStatefulSet()
+				sts.Spec.Template.Annotations[naming.ForceRedeploymentReasonAnnotation] = "reason"
+				return sts
+			}(),
+			expectedError: nil,
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
let's add t.Parallel() in this case, this should force the global vars out

Reply:
added t.Parallel()